### PR TITLE
Use PulpTemporaryFile

### PR DESCRIPTION
**Pull request by alikins**
Changes to use PulpTemporaryFile instead of Artifact as per:

https://pulp.plan.io/issues/6718
https://github.com/pulp/pulp_ansible/pull/335/files
https://github.com/pulp/pulpcore/pull/793/files

Issue: #330
Issue: #333

### CODE REVIEW

File: dev/docker-compose.yml [hunk 1]
```diff
@@ -116,6 +112,7 @@ services:
 
 
 volumes:
+  pulp: {}
   pulp_artifact: {}
```
Comment (awcrosby) [inline dev/docker-compose.yml:116]: Is `pulp_artifact` still used / mapped above?

File: galaxy_ng/app/tasks/publishing.py [hunk 2]
```diff
@@ -25,54 +42,67 @@ def import_and_move_to_staging(artifact_pk, **kwargs):
...
-    enqueue_with_reservation(add_content_to_repository, locks, args=task_args)
 
-    # enqueue task to remove collection_verion from inbound repo
     inbound_repo = AnsibleRepository.objects.get(pk=inbound_repository_pk)
-    locks = [inbound_repo]
-    task_args = (collection_version.pk, inbound_repository_pk)
-    enqueue_with_reservation(remove_content_from_repository, locks, args=task_args)
 
+    remove_locks = [staging_repo]
+    add_locks = [inbound_repo]
```
Comment (awcrosby) [inline galaxy_ng/app/tasks/publishing.py:56]: I think these two values need to be switched

File: galaxy_ng/app/api/v3/viewsets/collection.py [hunk 3]
```diff
@@ -140,10 +140,11 @@ class CollectionUploadViewSet(LocalSettingsMixin, pulp_ansible_views.CollectionU
     ]
     parser_classes = [AnsibleGalaxy29MultiPartParser]
 
-    def _dispatch_import_collection_task(self, artifact_pk, repository=None, **kwargs):
+    def _dispatch_import_collection_task(self, temp_file_pk, repository=None, **kwargs):
         """Dispatch a pulp task started on upload of collection version."""
-        locks = [str(artifact_pk)]
-        kwargs["artifact_pk"] = artifact_pk
+        locks = [str(temp_file_pk)]
```
Comment (fao89) [inline galaxy_ng/app/api/v3/viewsets/collection.py:145]: I believe temp_file does not need a lock, @bmbouter wdyt?

File: dev/docker-compose.yml (hunk 1 above)
Comment (alikins) [reply dev/docker-compose.yml:116]: It is not used now, removing.

File: galaxy_ng/app/tasks/publishing.py (hunk 2 above)
Comment (alikins) [reply galaxy_ng/app/tasks/publishing.py:56]: It certainly looks wrong now. Testing it again now to see if I switch it for a reason.

File: galaxy_ng/app/api/v3/viewsets/collection.py (hunk 3 above)
Comment (alikins) [reply galaxy_ng/app/api/v3/viewsets/collection.py:145]: pulp_ansible's impl of _dispatch_import_collection_task() doesn't seem to use.

That may be a left over from before I figured out the share /var/lib/pulp/ issues.
Seeing if I can remove it.

Comment (alikins) [reply galaxy_ng/app/api/v3/viewsets/collection.py]: Doesn't seem to need a lock. Updated.

**Pull request merged**